cmake/depend/pal.cmake

```diff
@@ -0,0 +1,30 @@
+message(STATUS "Adding dependency: pal")
+
+FetchContent_Declare(
+    pal
+    GIT_REPOSITORY  https://github.com/bareflank/pal.git
+    GIT_TAG         20.03.1
+)
+
+FetchContent_GetProperties(pal)
+
+if(NOT pal_POPULATED)
+    set(PAL_QUIET_CMAKE ON)
+    set(PAL_TARGET_LANGUAGE c++11)
+    set(PAL_OUTPUT_DIR ${CMAKE_BINARY_DIR}/depend/pal-generated-output)
+
+    if(${HYPERVISOR_TARGET_ARCH} STREQUAL x86_64)
+        set(PAL_TARGET_ARCH intel_x64)
+        set(PAL_ACCESS_MECHANISM gas_att)
+    elseif(${HYPERVISOR_TARGET_ARCH} STREQUAL AMD64)
+        set(PAL_TARGET_ARCH intel_x64)
+        set(PAL_ACCESS_MECHANISM gas_att)
+    elseif(${HYPERVISOR_TARGET_ARCH} STREQUAL armv8a)
+        set(PAL_TARGET_ARCH armv8a)
+        set(PAL_ACCESS_MECHANISM gas_aarch64)
+    endif()
+
+    FetchContent_Populate(pal)
+
+    add_subdirectory(${pal_SOURCE_DIR} ${pal_BINARY_DIR})
+endif()
```

Comment:
PAL seems to be integrating nicely in to the hypervisor repo using the FetchContent strategy. One thing to keep in mind: adding pal as a dependency to hypervisor comes with a few additional build dependencies

1) You must install Python > 3.6

2) You must ```pip3 install lxml dataclasses colorama pyyaml```

Are we ok with that?

Reply:
Glad its working out ok! Personally I'm good with the added dependencies. So going forward any of bareflank's cmake projects will need to consistently use a prefix so they dont clash with others?

Reply:
Yep, you got the idea! All cache variables and functions could follow a convention like \<PROJECT_NAME\>_\<CONFIG_NAME\>, and that would prevent any collisions from integration via FetchContent.

Even though its a little more verbose, a naming convention like that does make it really easy to glance which configs apply to which projects while viewing in ccmake/cmake-gui. The downside is you end up with quite a few variables that sorta look duplicated (e.g. all the projects have a concept of target_arch, a target_platform, a build_type, etc)